ksqldb-rest-app/src/main/java/io/confluent/ksql/api/spi/QueryPublisher.java

```diff
@@ -24,7 +25,7 @@
  * query that is executed. A subscriber from the API implementation then subscribes to it, then a
  * stream of query results flows from back-end to front-end where they are written to the wire.
  */
-public interface QueryPublisher extends Publisher<GenericRow> {
+public interface QueryPublisher extends Publisher<KeyValue<List<?>, GenericRow>> {
```

Comment:
How come we use `List<?>` for the key type and `GenericRow` for the value type throughout this PR? Isn't `GenericRow` essentially just a wrapper around `List<?>`? Wondering what motivates the inconsistency here.

Reply:
Not sure I follow...

I you suggesting we should have `KeyValue<GenericRow, GenericRow>`?  That would obviously be wrong, as the key shouldn't be represented by `GenericRow`.

Or maybe `KeyValue<List<?>, List<?>>`?  We could do this, but that's changing more code.  The PR just adds the key, (which is just a list of values), and leaves the value as it was.   We could switch the GenericRow to `List<?>`, but... out of scope...

Reply:
I think I see the difference now... while it's true that `GenericRow` essentially just wraps `List<?>`, the interpretation of the list elements in `GenericRow` is that each element represents a column, which is a dubious interpretation for the key since today the key is either a single column or three columns (single, primary column plus window start and end). Is that the reason you say `KeyValue<GenericRow, GenericRow>` would be incorrect?